crates/handlers/src/passwords.rs

```diff
@@ -35,6 +36,9 @@ struct InnerPasswordManager {
 
     /// A map of "old" hashers used only for verification
     other_hashers: HashMap<SchemeVersion, Hasher>,
+
+    /// The REST authentication provider URL, if any
+    rest_auth_provider: Option<RestAuthProviderConfig>,
 }
 
 impl PasswordManager {
```

Comment:
It feels like the password verification logic should be somewhere here instead of in the `compat` module

Reply:
You mean the whole call to the password provider (`authenticate_via_rest_api`)? This could be moved in the handlers create, of course, but then we should also move `user_login_with_password` there, no?